vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBox.java

```diff
@@ -711,9 +1082,41 @@ public void setDataProvider(ItemFilter<T> itemFilter,
         setDataProvider(listDataProvider,
                 filterText -> item -> itemFilter.test(item, filterText));
 
-        // Force the data communicator initialization eagerly because in-memory
-        // data is used (ListDataProvider)
-        initDataCommunicator();
+        // Enable fetch and size queries eagerly because in-memory
+        // data is used and getting the collection size is not so
+        // heavy like a database call
+        enableDataProviderFetch();
```

Comment:
This does eagerly something that was postponed in the call a few lines earlier - which is a code smell.

Reply:
Yes, looks ugly. I will add the check depending on DataProvider type

Reply:
Done